pyproject.toml

```diff
@@ -1,3 +1,156 @@
+
+[tool.setuptools.packages.find]
+include = ["e3sm_diags", "e3sm_diags.*"]
+
+[tool.setuptools.dynamic]
+version = { attr = "e3sm_diags.__version__" }
+
+[tool.setuptools.data-files]
+"share/e3sm_diags" = [
+  "e3sm_diags/driver/acme_ne30_ocean_land_mask.nc",
+  "misc/e3sm_logo.png",
+]
+
+"share/e3sm_diags/viewer" = ["e3sm_diags/viewer/index_template.html"]
+
+"share/e3sm_diags/zonal_mean_xy" = [
+  "e3sm_diags/driver/default_diags/zonal_mean_xy*",
+  "e3sm_diags/driver/default_diags/legacy_diags/zonal_mean_xy*",
+]
+"share/e3sm_diags/zonal_mean_2d" = [
+  "e3sm_diags/driver/default_diags/zonal_mean_2d_model*",
+  "e3sm_diags/driver/default_diags/legacy_diags/zonal_mean_2d*",
+]
+"share/e3sm_diags/zonal_mean_2d_stratosphere" = [
+  "e3sm_diags/driver/default_diags/zonal_mean_2d_stratosphere*",
+]
+"share/e3sm_diags/meridional_mean_2d" = [
+  "e3sm_diags/driver/default_diags/meridional_mean_2d*",
+]
+"share/e3sm_diags/lat_lon" = [
+  "e3sm_diags/driver/default_diags/lat_lon*",
+  "e3sm_diags/driver/default_diags/legacy_diags/lat_lon*",
+]
+"share/e3sm_diags/polar" = [
+  "e3sm_diags/driver/default_diags/polar*",
+  "e3sm_diags/driver/default_diags/legacy_diags/polar*",
+]
+"share/e3sm_diags/lat_lon_vector" = [
+  "e3sm_diags/driver/default_diags/lat_lon_vector*",
+]
+"share/e3sm_diags/lat_lon_land" = [
+  "e3sm_diags/driver/default_diags/lat_lon_land*",
+]
+"share/e3sm_diags/lat_lon_river" = [
+  "e3sm_diags/driver/default_diags/lat_lon_river*",
+]
+"share/e3sm_diags/cosp_histogram" = [
+  "e3sm_diags/driver/default_diags/cosp_histogram*",
+  "e3sm_diags/driver/default_diags/legacy_diags/cosp_histogram*",
+]
+"share/e3sm_diags/area_mean_time_series" = [
+  "e3sm_diags/driver/default_diags/area_mean_time_series*",
+]
+"share/e3sm_diags/enso_diags" = ["e3sm_diags/driver/default_diags/enso_*"]
+"share/e3sm_diags/qbo" = ["e3sm_diags/driver/default_diags/qbo*"]
+"share/e3sm_diags/streamflow" = ["e3sm_diags/driver/default_diags/streamflow*"]
+"share/e3sm_diags/diurnal_cycle" = [
+  "e3sm_diags/driver/default_diags/diurnal_cycle_*",
+]
+"share/e3sm_diags/arm_diags" = ["e3sm_diags/driver/default_diags/arm_diags_*"]
+"share/e3sm_diags/tc_analysis" = [
+  "e3sm_diags/driver/default_diags/tc_analysis_*",
+]
+"share/e3sm_diags/annual_cycle_zonal_mean" = [
+  "e3sm_diags/driver/default_diags/annual_cycle_zonal_mean_*",
+]
+"share/e3sm_diags/aerosol_aeronet" = [
+  "e3sm_diags/driver/default_diags/aerosol_aeronet*cfg",
+  "e3sm_diags/driver/default_diags/aerosol_aeronet_data/*.txt",
+]
+"share/e3sm_diags/aerosol_budget" = [
+  "e3sm_diags/driver/default_diags/aerosol_budget*cfg",
+]
+"share/e3sm_diags/mp_partition" = [
+  "e3sm_diags/driver/default_diags/mp_partition*cfg",
+]
+"share/e3sm_diags/tropical_subseasonal" = [
+  "e3sm_diags/driver/default_diags/tropical_subseasonal*cfg",
+]
+"share/e3sm_diags/colormaps" = ["e3sm_diags/plot/colormaps/*.rgb"]
+"share/e3sm_diags/control_runs" = ["e3sm_diags/driver/control_runs/*"]
+
```

Comment:
Hi @xylar, I thought it'd be a good idea to tag you for review here. I replaced the `setup.py` file with `pyproject.toml`. If you don't have time to review, it should be fine since the CI/CD build.